### PR TITLE
feature[javalib]: Math.TAU & Math.clamp() overloaded methods

### PR DESCRIPTION
**Pull request by LeeTibbert**
Just in time for Tau Day, 6.28.2024

When compiling using the corresponding Java version or above, two javalib `Math` features are
now available:

1) The constant `Math.TAU` is now evident in `Math.scala` - Java 19. 

    Previously the constant was available to user code[^1]  but did not appear in the file.
    Now it is easier for readers of that file to know that `Math.TAU` is supported.

2) The four overloaded variants of `Math.clamp()`  are implemented & tested - Java 21.

[^1]: An early pass of the  Scala Native compiler resolves symbols from the runtime Java library
       supplied (JVM), not the Scala Native javalib. `Math.TAU` is a final symbol, which means that
       it will be resolved and in-lined well before any javalib `Math.scala` lookup.  This means that
       the actual value in javalib `Math.scala` is irrelevant.  This PR adds the symbol so that
       human or AI readers do not need to know the intimate details of the Scala Native compiler
       in order to believe that the symbol is available to use. See one; do one.

### CODE REVIEW

**Comment by LeeTibbert** [comment]
Some of these failures are mine. I add two new tests for -0.0 handling and they
reported a defect in the underlying code. 

Converting to draft until I can fix that defect.

**Comment by LeeTibbert** [comment]
Perhaps Tau Day 2025, grrhh.

I am using LLVM 18 on my macOS system and both my  original code and the latest variant 
using another LLVM intrinsic pair (maximum & minimum) work just file.

The LLVM in CI chokes.  The "maximum" and "minimum" pair are supposed to be LLVM
at least as far back as Clang 8.  CI seems to disagree.

More tracing by light of day.

**Comment by LeeTibbert** [comment]
Today I learned:

The `llvm.minimum` and `llvm.maximum` intrinsics are not available with economic effort in 
the Clang version (14) used  in CI on macOS and Linux systems. It/they are also not available
in the Clang 10 used on arm-32.

They seem to be available & doing the job on Clang (15) used in Windows and arm-64 tests.

Of course, they function just well on my Linux & macOS development system.

TL;DR - can not use these intrinsics on 64 bit systems until the minimal Clang/LLVM version is
             greater than or equal to 15.

**Comment by LeeTibbert** [comment]
The scratch work in this draft PR has identified a number of Issues. It has well earned its keep.

My current plan is to wait until PR #3985 is merged. A comprehensive Test suite for the floating point
versions of `clamp()` require testing IEEE754 negative zeros, particularly as low and/or high bounds.
Are they the expected -0.0 or are they +0.0. 

Then I will, fate willing, create a new PR, based on the learnings from this one, with a short & tidy commit audit trail.
It is like the steps of a contra-dance: one-by-one, in the proper sequence.

**Comment by LeeTibbert** [comment]
Perhaps this PR should apply for a bug bounty or three?

**Comment by LeeTibbert** [comment]
Superseded by new PR #3989, with greatly benefited from the scratch work in this draft PR
